app/src/main/java/com/teamagam/gimelgimel/app/common/logging/Log4jDiskLogger.java

```diff
@@ -0,0 +1,221 @@
+package com.teamagam.gimelgimel.app.common.logging;
+
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+
+import java.io.File;
+
+import de.mindpipe.android.logging.log4j.LogConfigurator;
+
+
+class Log4jDiskLogger extends BaseLifecycleLogger {
+
+    private static final String VERBOSITY_USER_INTERACTION = "USER_INTERACTION";
+    private static final String VERBOSITY_LIFECYCLE = "LIFECYCLE";
+
+    private Logger mLogger;
+
+    Log4jDiskLogger(String directory, String fileName, long maxFileSize, int maxBackupLogFiles, String tag) {
```

Comment:
It seems that this object cannot function if it fails to instantiate the log4j.Logger.
I'd let the exception propagate upwards, for others to catch and handle.
This way, all those if(mLogger != null) checks can be eliminated and make the code much cleaner :)

I would make the constructor ask for a log4j.Logger and simply move the responsibility to a static favtory method. (easier to test that way, not that this class seems to need any testing).
!fix porfavor